src/huntsman/pocs/observatory.py

```diff
@@ -789,3 +514,127 @@ def _move_all_filterwheels_to(self, filter_name, camera_names=None):
+            self.logger.info(f'Slewing to flat field coordinates: {observation.field}.')
+            self.mount.set_target_coordinates(observation.field)
+            self.mount.slew_to_target()
+
+            # Get standard fits headers
+            headers = self.get_standard_headers(observation=observation)
+
+            # Start the exposures on each camera
+            events = {}
+            exptimes = {}
+            filenames = {}
+            start_times = {}
+            for cam_name, seq in sequences.items():
+                camera = self.cameras[cam_name]
+
+                # Get exposure time, filename and current time
+                exptimes[cam_name] = seq.get_next_exptime(past_midnight=self.past_midnight)
+                filenames[cam_name] = observation.get_exposure_filename(camera)
+                start_times[cam_name] = current_time()
+
+                # Start the exposure and get event
+                # TODO: Replace with concurrent.futures
+                events[cam_name] = camera.take_observation(
+                    observation, headers=headers, filename=filenames[cam_name],
+                    exptime=exptimes[cam_name])
+
+            # Wait for the exposures, dropping cameras that timeout
+            duration = get_quantity_value(max(exptimes.values()), u.second) + timeout
+            self._wait_for_camera_events(events, duration, remove=True)
+
+            # Update the flat field sequences with new data
+            for cam_name in list(sequences.keys()):
+                if cam_name not in self.cameras.keys():  # Camera removed
+                    del sequences[cam_name]
+                    continue
+                sequences[cam_name].update(filenames[cam_name], exptime=exptimes[cam_name],
+                                           time_start=start_times[cam_name])
+                # Log sequence status
+                status = sequences[cam_name].status
+                self.logger.info(f"Flat field status for {cam_name}: {status}")
+
+    def _autoflat_target_counts(self, cam_name, target_scaling, scaling_tolerance):
+        """ Get the target counts and tolerance for each camera.
+        Args:
+            cam_name (str): The camera name.
+            target_scaling (float):
+            scaling_tolerance (float):
+        """
+        camera = self.cameras[cam_name]
+        try:
+            bit_depth = camera.bit_depth.to_value(u.bit)
+        except NotImplementedError:
+            self.logger.debug(f'No bit_depth property for {cam_name}. Using 16.')
+            bit_depth = 16
+
+        target_counts = int(target_scaling * 2 ** bit_depth)
+        counts_tolerance = int(scaling_tolerance * 2 ** bit_depth)
+
+        self.logger.debug(f"Target counts for {cam_name}: {target_counts}"
+                          f" ± {counts_tolerance}")
+        return target_counts, counts_tolerance
+
+    def _wait_for_camera_events(self, events, duration, remove=False, sleep=1):
```

Comment:
Something more explicit than just `remove` is probably a bit better. `remove_camera`?
```suggestion
    def _wait_for_camera_events(self, events, duration, remove_on_error=False, sleep=1):
```